### PR TITLE
Determine the lastCompletelyProcessedLsn from commitRecord() instead of poll()

### PR DESCRIPTION
**Pull request by mmarvick-convoy**

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java [hunk 1]
```diff
@@ -159,13 +161,30 @@ public void commit() throws InterruptedException {
         }
     }
 
+    @Override
+    public void commitRecord(SourceRecord record) throws InterruptedException {
+        if (running.get()) {
```
Comment (mwinstanley) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java]: I don't think you need to bother with this check; there's no side effect where it matters, commit() will deal with it.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java [hunk 2]
```diff
@@ -159,13 +161,30 @@ public void commit() throws InterruptedException {
...
+            Long recordLsn = (Long) record.sourceOffset().get(SourceInfo.LSN_KEY);
+
+            // If this record has an lsn and it's larger than the last seen lsn, then we've fully processed
+            // the events with the last seen lsn
+            if (recordLsn != null && recordLsn > previousLsn) {
```
Comment (mwinstanley) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java]: I think you need to check that previousLsn is not null too

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java [hunk 3]
```diff
@@ -159,13 +161,30 @@ public void commit() throws InterruptedException {
...
+                // Update the last completely processed lsn to the previous batch's lsn
+                lastCompletelyProcessedLsn = previousLsn;
+
+                // Update previous lsn for comparison when processing the next batch
+                previousLsn = recordLsn;
+            }
```
Comment (mwinstanley) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java]: Can we add a log line of some kind to see if we ever see a previousLsn > recordLsn?

**Review by mwinstanley — APPROVED** [review]
Just one correctness issue - rest are nits

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java (hunk 1 above)
Comment (mmarvick-convoy) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java]: Agreed :+1:

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java [hunk 4]
```diff
@@ -45,6 +45,8 @@
     private PostgresTaskContext taskContext;
     private RecordsProducer producer;
 
+    private Long previousLsn;
```
[the remark this answers is not in the document]
Comment (mmarvick-convoy) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java]: Good catch, thanks

**Pull request merged**